diff -uNr --no-dereference a/libxtables/Makefile.in b/libxtables/Makefile.in
--- a/libxtables/Makefile.in	2022-05-13 15:49:58.000000000 +0200
+++ b/libxtables/Makefile.in	2022-10-15 18:54:08.000000000 +0200
@@ -362,7 +362,7 @@
 top_srcdir = @top_srcdir@
 xtlibdir = @xtlibdir@
 AM_CFLAGS = ${regular_CFLAGS}
-AM_CPPFLAGS = ${regular_CPPFLAGS} -I${top_builddir}/include -I${top_srcdir}/include -I${top_srcdir}/iptables ${kinclude_CPPFLAGS}
+AM_CPPFLAGS = ${regular_CPPFLAGS} -I${top_builddir}/include -I${top_srcdir}/include -I${top_srcdir}/iptables -I${top_srcdir} ${kinclude_CPPFLAGS}
 lib_LTLIBRARIES = libxtables.la
 libxtables_la_SOURCES = xtables.c xtoptions.c getethertype.c
 libxtables_la_LDFLAGS = -version-info ${libxtables_vcurrent}:0:${libxtables_vage}
